### DISCUSSION TITLE
How to delete all rules from root level.

### DISCUSSION THREAD

**Discussion by viveknandhangelella** [react-querybuilder/react-querybuilder#534] [Q&A] [answered]
Hi @jakeboone02 , 

Do we have any option to delete at root level so that all rules are removed

Thanks,
Vivek

**Comment by jakeboone02** [comment]
There's no standard way to do it, but depending on how you're managing the query it's really just a matter of emptying the `rules` array at the root of the object.

**Reply by viveknandhangelella** [reply]
Hi @jakeboone02  
could you please help me to add button same as delete done at group level
thanks,
Vivek

**Reply by jakeboone02** [reply, marked as answer]
Do you want the button to appear outside of or within the boundaries of the `<QueryBuilder>` component? If outside, just set your query as you do in the `onQueryChange` handler. Something like this:

```tsx
const App = () => {
  const [query, setQuery] = useState(initialQuery);

  const resetQuery = () => setQuery(initialQuery);

  return (
    <div>
      <button onClick={resetQuery}>Remove all rules</button>
      <QueryBuilder query={query} onQueryChange={q => setQuery(q)} />
    </div>
  );
};
```

If you need the button to be _within_ the boundaries of the component, see https://github.com/react-querybuilder/react-querybuilder/issues/533#issuecomment-1599744219.

**Reply by viveknandhangelella** [reply]
Hi @jakeboone02 , 

Thank you I would like to add within border. above solution worked perfectly. 
Need another help on how to add custom component to rule that component alter query by adding additional property

eg:- 
A check box at rule 
if I add check then rule will have active property which will be changed based on user action 

![Screenshot 2023-07-31 at 4 31 04 PM](https://github.com/react-querybuilder/react-querybuilder/assets/36690239/9f844e15-6cbb-4856-8f4a-26dd2b246b31)

rule:  rules: [{
    rule: {
      field: 'availableForuse',
      operator: 'EQUAL',
      value: [true],
      active: true,
    }
  }]

active value is based on user selecting checkbox
